kiwi/bootloader/config/systemd_spec_base.py

```diff
@@ -0,0 +1,211 @@
+# Copyright (c) 2022 Marcus Schäfer
+#
+# This file is part of kiwi.
+#
+# kiwi is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# kiwi is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with kiwi.  If not, see <http://www.gnu.org/licenses/>
+#
+from typing import (
+    Dict, NamedTuple
+)
+
+# project
+from kiwi.bootloader.config.base import BootLoaderConfigBase
+
+target_type = NamedTuple(
+    'target_type', [
+        ('disk', str),
+        ('install', str),
+        ('live', str)
+    ]
+)
+
+
+class BootLoaderSystemdSpecBase(BootLoaderConfigBase):
+    """
+    **Base class for Bootloader Spec configuration**
```

Comment:
I think it'd be better to spell this "Boot Loader Specification" to match the spec docs.

FWIW, the specification was very much designed to be independent of any implementation. So "systemd" is **not** part of the name. We nowadays say add "`Type #1`" or "`Type #2`" to distinguish unified kernels where appropriate.

Confusion was created by grub2 implementing something incompatible and using the same name. But those patches are only downstream in Fedora/RHEL and maybe won't even be merged anywhere. So I think it'd be better to distinguish that special variant by some special name, "GRUB2 BLS-based template language" or something, if it is ever implemented here.

Reply:
Strictly speaking, that template stuff are supposed to be mostly gone now...

Reply:
I think that apart from the syntax changes, the other differences are not important. Grub2 used some different locations for files, but that kind of thing is better handled through a configuration parameter, not a separate class.

Reply:
Thanks, I applied the proposed changes on the naming.

grub2 BLS will be a matter of a new pull request. With this one the main interest is to get the API right to handle BLS compatible loaders and systemd-boot will be the, let's say, first implementation :)

Thanks